llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -824,7 +824,13 @@ namespace llvm {
 Value *createStepForVF(IRBuilderBase &B, Type *Ty, ElementCount VF,
                        int64_t Step) {
   assert(Ty->isIntegerTy() && "Expected an integer step");
-  return B.CreateElementCount(Ty, VF.multiplyCoefficientBy(Step));
+  if (!VF.isScalable() || !isPowerOf2_64(VF.getKnownMinValue()) ||
```

Comment:
nit: Do we need the check for `isPowerOf2_64(VF.getKnownMinValue())`? I thought the vectoriser rejected non-power-of-2 VFs? I just wonder if this can be an assert instead.